app/models/patient.rb

```diff
@@ -646,7 +646,11 @@ def sessions
   end
 
   def archived?(team:)
-    archive_reasons.exists?(team:)
+    if archive_reasons.loaded?
+      archive_reasons.select { it.team_id == team.id }.any?
```

Comment:
```suggestion
      archive_reasons.any? { it.team_id == team.id }
```

Probably won't make much of a difference in production as there won't be many archive reasons, but calling `any?` like this will ensure that it returns `true` on the first one it finds short circuiting the rest.

Reply:
Ah, yes, thanks again! I had caught that in `not_in_team?` but not here.